### ISSUE TITLE
[Kagura] Wi-Fi out of memory in driver

### ISSUE THREAD

**Issue by Revertron** [sonyxperiadev/bug_tracker#440]
**Platform:** Tone
**Device:** Kagura
**Kernel version:** 4.9.182-gf79dfb7cd4d3
**Android version:** 9 Pie

**Description**
Wi-Fi stops working at all, stuck on searching for networks.
```
2019-07-09 14:36:06.434 1126-3495/? I/WifiService: startScan uid=1000
2019-07-09 14:36:06.442 1126-1229/? E/WificondScannerImpl: Failed to start scan, freqs=null
2019-07-09 14:36:06.443 1126-1226/? E/WifiScanRequestProxy: Scan failure received. reason: -1,description: Scan failed
2019-07-09 14:36:06.450 7650-7650/? D/WifiSettings: onAccessPointsChanged (WifiTracker) callback initiated
2019-07-09 14:36:06.441 703-703/? E/wificond: NL80211_CMD_TRIGGER_SCAN failed: Out of memory
```

**Symptoms**
Wi-Fi does not show any networks, always searching. Only reboot helps.

**How to reproduce**
Enable Wi-Fi, connect to any network, use as normal. After some time go somewhere far from router, walk for some time (maybe some hour or two) and come back. Wi-Fi can't connect back, can't find any network.

[logcat.txt](https://github.com/sonyxperiadev/bug_tracker/files/3387666/logcat.txt)

These lines in logcat are repeating on and on.

**Comment by YuriyMosiychuk** [comment]
+1
The same problem in suzu, loire platform.

**Comment by kholk** [comment]
That's an interesting one. I have something that will probably alleviate/posticipate this issue (because it was made for another issue...) and it's currently under testing.

Anyway, can you please "cat /proc/meminfo" whenever this happens again (in the same boot, of course)? It would be interesting to know what's going on with your RAM :)

**Comment by Revertron** [comment]
I've repeated the bug, here is the meminfo:
```
kagura:/ $ cat /proc/meminfo
MemTotal:        2700568 kB
MemFree:           43012 kB
MemAvailable:     635916 kB
Buffers:           32252 kB
Cached:           647216 kB
SwapCached:         2796 kB
Active:          1109040 kB
Inactive:         597016 kB
Active(anon):     825408 kB
Inactive(anon):   299600 kB
Active(file):     283632 kB
Inactive(file):   297416 kB
Unevictable:       90448 kB
Mlocked:           90448 kB
SwapTotal:        255996 kB
SwapFree:         191644 kB
Dirty:                 0 kB
Writeback:             0 kB
AnonPages:       1116240 kB
Mapped:           532644 kB
Shmem:              8548 kB
Slab:             191604 kB
SReclaimable:      49880 kB
SUnreclaim:       141724 kB
KernelStack:       32048 kB
PageTables:        44356 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     1606280 kB
Committed_AS:   64396896 kB
VmallocTotal:   263061440 kB
VmallocUsed:           0 kB
VmallocChunk:          0 kB
CmaTotal:         151552 kB
CmaFree:               0 kB
```

**Comment by kholk** [comment]
Oh that's interesting. So the CMA is totally exhausted.
Great to know, we'll think about testing some stuff soon.

**Comment by jerpelea** [comment]
@Revertron updates on this ?
there were serveral fixes since the bug was posted but there is no followup

**Comment by Revertron** [comment]
@jerpelea did these fixes go to the one usable kernel (4.9)?
Because, the 4.14 is unusable.

**Comment by jerpelea** [comment]
there are backports from 4.14 to 4.9
